web/src/app/util/ISOPickers.js

```diff
@@ -0,0 +1,139 @@
+import React, { useState, useEffect } from 'react'
+import { DatePicker, TimePicker, DateTimePicker } from '@material-ui/pickers'
+import { useSelector } from 'react-redux'
+import { urlParamSelector } from '../selectors'
+import { DateTime } from 'luxon'
+import { TextField, InputAdornment, IconButton } from '@material-ui/core'
+
+import Modernizr from '../../modernizr.config'
+import { DateRange } from '@material-ui/icons'
+
+function hasInputSupport(name) {
+  if (new URLSearchParams(location.search).get('nativeInput') === '0') {
+    return false
+  }
+
+  return Modernizr.inputtypes[name]
+}
+
+function useISOPicker(
+  { value, onChange, timeZone, ...otherProps },
+  { format, truncateTo, type, Fallback },
+) {
+  const native = hasInputSupport(type)
+  const params = useSelector(urlParamSelector)
+  const zone = timeZone || params('tz', 'local')
+  const dtValue = DateTime.fromISO(value, { zone })
+  const [inputValue, setInputValue] = useState(dtValue.toFormat(format))
+
+  const parseInput = input => {
```

Comment:
Would `isoToDateTime` be a more accurate name?

Reply:
I think we want to keep `parse` as it's taking in raw input, in one of many possible formats, and we've been mostly consistent in naming things like that. But the DateTime output is important, I'll try coming up with something for that.